content/en/docs/armory-agent/agent-mtls.md

```diff
@@ -0,0 +1,267 @@
+      - name: certpem
+        secret:
+          secretName: ??? @TODO what secret is this?
+{{< /prism >}}
+
+
+### Configure the service
+
+@TODO what file is this in?
+
+{{< prism lang="yaml" >}}
+clouddriver:
+  grpc: <:443
+  insecure: false
+  tls:
+    #serverName: my-ca  #to override the server name to verify
+    insecureSkipVerify: false #if true, don't verify server's cert
+    clientKeyFile: /opt/kubesvc/cert/agent.key #ref to the private key (mTLS)
+    #clientKeyPassword: #if the clientKeyFile is password protected
+    #cacertFile: /opt/kubesvc/cacert/ca.crt #to validate server's cert
+    clientCertFile: /opt/kubesvc/cert/agent.crt #client cert for mTLS.
+
+{{< /prism >}}
+
+See the {{< linkWithTitle "agent-options.md" >}} for additional options.
+
+## Forward proxy configuration
+
+### Add proxy variables
+
+Add proxy variables in the Agent's deployment file. @TODO what file is this?
+
+{{< prism lang="yaml" line="25-30" >}}
+spec:
+  template:
+    spec:
+      containers:
+      - image: armory/kubesvc
+        env:
+        - name: HTTP_PROXY
+          value: <proxyaddress:proxyport>
+        - name: HTTPS_PROXY
+          value: <proxyaddress:proxyport>
+{{< /prism >}}
+
+### Configure the Agent
+
+Configure the Agent to **not** proxy Kubernetes traffic intended for KubeAPI in your cluster.
+
+Add the `noProxy: true` option to have traffic destined for Kubernetes ignore the proxy environment settings under `kubernetes`.
+
+@TODO what file is this?
+
+{{< prism lang="yaml" line="12" >}}
+kubernetes:  
+  accounts: []
+  noProxy: true
+{{< /prism >}}
+
+## x509 certificate subject filtering
+
+If you have many Agents that want to talk to Clouddriver, and all of them have valid x509 certificates for mTLS, you can authorize a particular subset by configuring a subject filter in your `clouddriver.yaml` configuration. If a certificate subject line matches **any** of the filters, that certificate is authorized. All non-matching certificate subjects calls are rejected with an `X509CertificateAuthorizationFilterException`.
+
+You can specify multiple filtering criteria. However, the order in which the criteria are read is not guaranteed. Be careful when matching on two parts of a subject line, for example `UID=.*O=Armory`, because the `UID` and `O` attributes may not appear in that order. It might be safest to write a regular expression that can match in any order.
+
+### Plugin filter configuration
+
+Add an `grpc.auth.x509` section to your Clouddriver profile. See lines 7-13 in the following example:
+
+
+{{< prism lang="yaml" line="7-13" >}}
+spec:
+  spinnakerConfig:;;
+    profiles:
+      clouddriver:
+        kubesvc:
+          cluster: redis
+          grpc:
+            auth:
+              x509:
+				        #Note: enabled must be true for filters to be applied
+                enabled: true
+                filters:
+                  - UID=([a-z]){3}:[1-9]{3}:ksvc
```

Comment:
Just FYI,  this should also be added to the document for clarity: https://docs.armory.io/docs/armory-agent/agent-plugin-options/ per the ticket https://armory.atlassian.net/browse/RBK-116

`kubesvc.grpc.auth.x509.enabled` and `kubesvc.grpc.auth.x509.filters`